ingestclient/core/engine.py

```diff
@@ -275,11 +283,25 @@ def run(self):
             raise Exception(msg)
 
         # Do some work
+        self.access_denied = False
+        self.access_denied_count = 0
+        self.invalid_access_key = False
+        self.invalid_access_key_count = 0
+
         wait_cnt = 0
         while True:
+            if self.access_denied:
+                self.access_denied = False
+                self.credential_create_time = datetime.datetime.min
+            if self.invalid_access_key:
+                self.invalid_access_key = False
+                if self.invalid_access_key_count % 5 == 4:
+                    self.credential_create_time = datetime.datetime.min
             # Check if you need to renew credentials
-            if (datetime.datetime.now() - self.credential_create_time).total_seconds() > self.backend.credential_timeout:
-                logger.warning("(pid={}) Credentials are expiring soon, attempting to renew credentials".format(os.getpid()))
+            total_seconds = (datetime.datetime.now() - self.credential_create_time).total_seconds()
+            if total_seconds > self.backend.credential_timeout:
+                logger.warning("(pid={}) Credentials are expiring soon, attempting to renew credentials".format(
```

Comment:
We could replace all of the references to `(pid={})` with `(pid=%(process)d)` in the formatter used for the application.

Ref: https://docs.python.org/3.6/library/logging.html#logrecord-attributes

Reply:
Good to know for the future.  I am going to leave it like this in the code.  It is this way several other locations in the file, this matches those examples.